tools/diff-processor/rules/message_context.go

```diff
@@ -3,21 +3,34 @@ package rules
 import (
 	"fmt"
 	"strings"
+
+	"github.com/GoogleCloudPlatform/magic-modules/tools/diff-processor/constants"
 )
 
 // MessageContext - is an envelope for additional
 // metadata about context around the rule breakage
 type MessageContext struct {
 	Resource   string
 	Field      string
+	definition string
+	name       string
 	identifier string
 	message    string
 }
 
-func populateMessageContext(message string, mc MessageContext) string {
+func populateMessageContext(message string, mc MessageContext) *BreakingChange {
+	template := message
 	resource := fmt.Sprintf("`%s`", mc.Resource)
 	field := fmt.Sprintf("`%s`", mc.Field)
 	message = strings.ReplaceAll(message, "{{resource}}", resource)
 	message = strings.ReplaceAll(message, "{{field}}", field)
-	return message + documentationReference(mc.identifier)
+	return &BreakingChange{
+		Resource:               mc.Resource,
+		Field:                  mc.Field,
+		Message:                message,
+		RuleTemplate:           template,
+		DocumentationReference: constants.GetFileUrl(mc.identifier),
+		RuleDefinition:         mc.definition,
+		RuleName:               mc.name,
```

Comment:
Do these need to be populated in IsRuleBreak?

Reply:
This function runs as part of the isRuleBreak. 
I included these because previously u said: "I'm not sure whether we will want to keep the name / definition fields long-term and there's not a clear need for them at the moment, but it probably wouldn't hurt to include them either"